### PR TITLE
feat: add `Set.eqOn_image`

### PR DESCRIPTION
**Pull request by urkud**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/Set/Function.lean [hunk 1]
```diff
@@ -208,6 +208,8 @@ theorem eqOn_comm : EqOn f₁ f₂ s ↔ EqOn f₂ f₁ s :=
 theorem eqOn_refl (f : α → β) (s : Set α) : EqOn f f s := fun _ _ => rfl
 #align set.eq_on_refl Set.eqOn_refl
 
+lemma eqOn_image : EqOn g₁ g₂ (f '' s) ↔ EqOn (g₁ ∘ f) (g₂ ∘ f) s := ball_image_iff
```
Comment (eric-wieser) [inline Mathlib/Data/Set/Function.lean:211]: This is practically the same lemma as `eqOn_comp_right_iff`; if you want both, can you put them next to each other?

Comment (urkud) [reply Mathlib/Data/Set/Function.lean:211]: No, I don't. I'm closing this PR. Thanks!

**Comment by urkud** [comment]
The lemma is a duplicate (with LHS and RHS swapped).

File: Mathlib/Data/Set/Function.lean (hunk 1 above)
Comment (eric-wieser) [reply Mathlib/Data/Set/Function.lean:211]: I think an alias might be appropriate here for discoverability; or renaming to `eqOn_comp_right_iff_eqOn_image`

**Pull request closed without merging**